### DISCUSSION TITLE
Active account from API

### DISCUSSION THREAD

**Discussion by OlegSokurData** [ory/kratos#3158] [Q&A]
Hi all. 
I have issue with account activation.

Steps
1) User pass data on sign-up from and send to our server.
2) Our server send to Kratos via updateRegistrationFlow
3) Then via webhook Kratos notify our service that new user appear
4) This service via Kratos identity API create recovery link and send it via email

Question: 
How we can complete recovery(account activation) flow via our service ? What kind of API we should you from our server ? How we can avoid csrf token ?

**Comment by vinckr** [comment]
Do you have a mobile/native or a browser client? 
These are the relevant APIs:
https://www.ory.sh/docs/reference/api#tag/frontend/operation/createBrowserRecoveryFlow
https://www.ory.sh/docs/reference/api#tag/frontend/operation/createNativeRecoveryFlow
https://www.ory.sh/docs/reference/api#tag/frontend/operation/updateRecoveryFlow
The CSRF token is required in browser flows, see: https://www.ory.sh/docs/kratos/concepts/security
